src/accelerate/accelerator.py

```diff
@@ -1423,7 +1423,10 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
 
                 # Check if the model is already a FSDP model due to `Manual Wrapping` and if so,
                 # don't wrap it again
-                if type(model) != FSDP:
+                is_torch_compiled = getattr(model, "_orig_mod", None)
```

Comment:
Does this work if we use `accelerate.utils.other.is_compiled_module`? I'd rather us update that function to check `_orig_mod` if it does not as well, since we have a util for doing so.

Reply:
Done.